### DISCUSSION TITLE
LanguageIds for uno-css language server in vscode

### DISCUSSION THREAD

**Discussion by parischap** [unocss/unocss#2886] [Q&A]
Hi there,

I am using unocss with preact.
When I use preact with TSX, everything works fine : the VS Code unocss extension will highlight my embedded utilities in .tsx files.
When I try to do the same with htm that replaces TSX, everything works fine (I mean my code works as expected), but I get no highligting and no autocompletion in my .ts files while developing. My feeling is that the vscode unocss extension does not watch .ts files because they are not supposed to contain embedded css. I tried to add `"unocss.languageIds": ["typescript"]` in my settings.json, but it does not change anything.

Anybody has any idea how to get that working?

Thanks in advance for your help.

**Comment by userquin** [comment]
try adding `ts` extension to the pipeline: https://unocss.dev/guide/extracting#extracting-from-build-tools-pipeline

**Reply by parischap** [reply]
Works perfectly thanks !

**Reply by userquin** [reply]
Maybe you can just add this comment to your ts files if there are only a few:
```ts
// @unocss-include
```

**Reply by parischap** [reply]
Yes I have seen that too now.
Only issue I have is transformerAttributifyJsx does not work any longer for the same reason: it does not look at ts files...

**Reply by parischap** [reply]
I figured out how to solve that one by myself. There is an include option on the transformer!
